Storage.md

```diff
@@ -0,0 +1,214 @@
+To save all the items' values to localStorage, use `saveAll`.
+
+```typescript
+ItemsHolder.saveItem("color");
+ItemsHolder.saveAll();
+```
+
+`increase` and `decrease` add and subtract respectively from the item's value.
+`toggle` flips the boolean value of the item.
+
+```typescript
+ItemsHolder.increase("weight", 14);
+ItemsHolder.decrease("weight", 10);
+ItemsHolder.toggle("started");
+```
+
+`checkExistence` can be used to check if there is an item under the specified key and if not, add it to the collection.
+
+```typescript
+ItemsHolder.checkExistence("color");
+```
+
+By default if an item does not exist when `setItem`, `getItem`, `increase`, or `toggle` are called, the item is then added to the collection.
+Define `allowNewItems` to explicitly enable or disable this feature.
+
+```typescript
+let ItemsHolder: ItemsHoldr = new ItemsHoldr({ allowNewItems: false });
+```
+
+## Updating
+
+Triggers are functions that get run when an item's value equals the specified value.
+
+```typescript
+ItemsHolder.addItem("color", {
+    value: "cyan",
+    triggers: {
+        red: () => { this.value = "black"; }
+    }
+});
+ItemsHolder.setItem("color", "red");
+```
+
+Modularity restricts an item's value to a range from 0 to the specified max and keeps running the `onModular` function until the value hits the modular range.
+
+```typescript
+ItemsHolder.addItem("counter", { value: 0 });
+ItemsHolder.addItem("time", {
+    value: 0,
+    modularity: 12,
+    onModular: () => { ItemsHolder.increase("counter", 1); }
+});
+ItemsHolder.setItem("time", 100);
+```
+
+### Auto Save
+
+By default, the values in localStorage are not updated when their values change.
+Auto saving, which updates a value in localStorage, can be enabled when making the ItemsHolder container.
+
+```typescript
+let ItemsHolder = new ItemsHoldr({ autoSave: true });
+```
+
+To toggle autoSave, use `toggleAutoSave`.
+
+```typescript
+ItemsHolder.toggleAutoSave();
+```
+
+## Clearing and Defaults
+
+To clear all items from the collection, use `clear`.
+
+```typescript
+ItemsHolder.clear();
+```
+
+To clear the collection, but keep some items which will always be in the collection, use `valueDefault`.
+
+```typescript
+let ItemsHolder: ItemsHoldr = new ItemsHoldr({
+    values: {
+        color: {
+            valueDefault: "red"
+        }
+    }
+});
+ItemsHolder.clear(); // "color" is still in the collection with a reset value of "red"
+```
+
+## Elements
+
+*Note:* This part of ItemsHoldr will be placed into another module in the future.
+
+To signal if a container should be made to hold HTML elements, assign a value to `doMakeContainer`.
+
+```typescript
+let ItemsHolder: ItemsHoldr = new ItemsHoldr({ doMakeContainer: true });
+```
+
+With this, any HTML items passed in at the time of construction are held in ItemsHolder's `container` object.
+HTML elements stored within ItemsHoldr will have their values on the page updated if the container is appended to an element on the page.
```

Comment:
I'm pretty sure the HTML values are still updated even if the container is never appended to an element on the page.